modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileDataStore.java

```diff
@@ -133,7 +133,13 @@ public ShapefileDataStore(URL url) {
     }
 
     Name getTypeName() {
-        return new NameImpl(namespaceURI, shpFiles.getTypeName());
+        String typeName = "Null";
```

Comment:
How did this happen?

Reply:
Not sure, possibly while debugging an earlier effort, reverted.

Reply:
Sometimes IndexManager.hasFidIndex() tries to get a lock using a store (in the IndexManager) where shpFiles is null. 

It seems to be if you ask for a FeatureIterator from a FeatureCollection that points to a DataStore that doesn't exist anymore.  (See line 222-223 in FidQueryTest).

Reply:
Roger that